bin/metrics-asg.rb

```diff
@@ -114,33 +107,34 @@ def cloud_watch_metric(metric_name, value, asg_name)
 
   def print_statistics(asg_name, statistics)
     result = {}
-    static_value = {}
     statistics.each do |key, static|
       r = cloud_watch_metric(key, static, asg_name)
-      static_value['AutoScalingGroup.' + asg_name + '.' + key + '.' + static] = static
-      result['AutoScalingGroup.' + asg_name + '.' + key + '.' + static] = r[:datapoints][0] unless r[:datapoints][0].nil?
+      keys =
+        if config[:scheme].nil?
+          []
+        else
+          [config[:scheme]]
+        end
+      keys.concat ['AutoScalingGroup', asg_name, key, static]
+
+      result[keys.join('.')] = r[:datapoints][0] unless r[:datapoints][0].nil?
```

Comment:
I personally prefer `.first` over `[0]` as it is a bit more expressive but I certainly won't make you change it if you don't want to.